zap/src/main/java/org/parosproxy/paros/Constant.java

```diff
@@ -1092,6 +1095,29 @@ private static void upgradeFrom2_8_0(XMLConfiguration config) {
         // Update to a newer default user agent
         config.setProperty(
                 ConnectionParam.DEFAULT_USER_AGENT, ConnectionParam.DEFAULT_DEFAULT_USER_AGENT);
+        updatePscanTagMailtoPattern(config);
+    }
+
+    private static void updatePscanTagMailtoPattern(XMLConfiguration config) {
+        String autoTagScannersKey = "pscans.autoTagScanners.scanner";
+        List<HierarchicalConfiguration> tagScanners = config.configurationsAt(autoTagScannersKey);
+        String badPattern = "<.*href\\s*['\"]?mailto:";
+        String goodPattern = "<.*href\\s*=['\"]?mailto:";
+
+        for (int i = 0, size = tagScanners.size(); i < size; ++i) {
+            String elementBaseKey = autoTagScannersKey + "(" + i + ").";
+            String currentKeyResBodyRegex = elementBaseKey + "resBodyRegex";
+            if (config.getProperty(currentKeyResBodyRegex).equals(badPattern)) {
+                config.setProperty(currentKeyResBodyRegex, goodPattern);
+                break;
+            }
+        }
+
+        try {
+            config.save();
```

Comment:
I've probably overlooked something, but this seemed necessary to make it stick.

Reply:
It shouldn't, the config is already being saved after calling all `upgradeFrom_` methods (when setting the latest version). I also tested without this statement and the tag was updated in the file.

Reply:
But did you see it in the UI as updated?

Reply:
Yes, it was the new value.

Reply:
Probably because of how I was testing? Inserting the method call at 610'ish so it'd be used in devmode... I'll remove this try block.

Reply:
Right, in that case it would not be saved.